axis_backend/apps/contracts/serializers/contract_serializer.py

```
@@ -0,0 +1,365 @@
+        # Validate dates if both are being updated or one is being updated
+        if 'start_date' in data or 'end_date' in data:
+            start_date = data.get('start_date', instance.start_date if instance else None)
+            end_date = data.get('end_date', instance.end_date if instance else None)
+
+            if start_date and end_date and end_date <= start_date:
+                raise serializers.ValidationError({
+                    'end_date': 'End date must be after start date'
+                })
+
+        # Validate renewal date if present
+        if 'renewal_date' in data and data['renewal_date']:
```

Ruff (0.14.5)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)